### PR TITLE
9571-Using-Clipboard-Tools-does-not-work-aka-Virtual-Paste

### LINKED ISSUE #9571: Using Clipboard Tools does not work (a.k.a Virtual Paste)
Using a tool to handle the clipboard does not work with P9.
These tools are implemented by sending a virtual paste event. 
This is usually implemented by sending a virtual Ctrl/Cmd + V event.

These events are handled by SDL, but they are handled by sending three events.

We need to correctly detect this sequence and handling the event.

### PR DESCRIPTION
**Pull request by tesonep**
Fix #9571

### CODE REVIEW

File: src/OSWindow-SDL2/OSSDLPasteEventFilter.class.st [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+		ifTrue: [ 
+			self addEvent: aSDLEvent.
+			self currentState: 2.
+			 ^ self ].
+
+	(self currentState = 1 and: [aSDLEvent isClipboardUpdateEvent]) 
+		ifTrue: [ 
+			self addEvent: aSDLEvent.
+			self currentState: 2.
+			 ^ self ].
```
Comment (guillep) [inline src/OSWindow-SDL2/OSSDLPasteEventFilter.class.st]: Aren't these two cases the same?

File: src/OSWindow-SDL2/OSSDLPasteEventFilter.class.st [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+OSSDLPasteEventFilter >> dispatchEvent: aSDLEvent [
+	
+	"This filter will handle the case when an external Clipboard handling tool is used.
+	When an external tool modifies the clipboard and send a virtual paste command, it sends three events:
+	
+	1) SDL_CLIPBOARDUPDATEEVENT
+	2) SDL_KEYDOWN with $v as the key
+	3) SDL KEYUP with $v as the keysym 
+	
+	If we find this three events, we need to modify the last two events to include defaul system modifier.
```
Comment (guillep) [inline src/OSWindow-SDL2/OSSDLPasteEventFilter.class.st]: this three events => these three events

File: src/OSWindow-SDL2/OSSDLPasteEventFilter.class.st [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+{ #category : #accessing }
+OSSDLPasteEventFilter >> currentState: aValue [
+
+	currentState := aValue
+]
+
+{ #category : #'events-processing' }
+OSSDLPasteEventFilter >> dispatchEvent: aSDLEvent [
+	
+	"This filter will handle the case when an external Clipboard handling tool is used.
```
Comment (guillep) [inline src/OSWindow-SDL2/OSSDLPasteEventFilter.class.st]: I don't know if it is worth to note that this seems to be a OSX SDL2 bug.

File: src/OSWindow-SDL2/OSSDLPasteEventFilter.class.st (hunk 1 above)
Comment (tesonep) [reply src/OSWindow-SDL2/OSSDLPasteEventFilter.class.st:93]: Yes, but I write it like this to make it more clear.

File: src/OSWindow-SDL2/OSSDLPasteEventFilter.class.st (hunk 3 above)
Comment (tesonep) [reply src/OSWindow-SDL2/OSSDLPasteEventFilter.class.st]: Yes, I will add that

File: src/OSWindow-SDL2/OSSDLPasteEventFilter.class.st (hunk 1 above)
Comment (tesonep) [reply src/OSWindow-SDL2/OSSDLPasteEventFilter.class.st]: Jajja, it is duplicated... I will fix it
 :D

**Comment by MarcusDenker** [comment]
```
Error
the following classes have unused instance variables and should be cleaned: an OrderedCollection(OSSDL2Driver)
Stacktrace
TestFailure
the following classes have unused instance variables and should be cleaned: an OrderedCollection(OSSDL2Driver)
NoUnusedVariablesLeftTest(TestAsserter)>>assert:description:resumable:
NoUnusedVariablesLeftTest(TestAsserter)>>assert:description:
NoUnusedVariablesLeftTest>>testNoUnusedInstanceVariablesLeft
NoUnusedVariablesLeftTest(TestCase)>>performTest
```

**Pull request merged**